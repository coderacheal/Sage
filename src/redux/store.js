import { configureStore } from '@reduxjs/toolkit';
import currencyReducer from './currencySlice';

const store = configureStore({
  reducer: {
    crypto: currencyReducer,
  },
});

export default store;
