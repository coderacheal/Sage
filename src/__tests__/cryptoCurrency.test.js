import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CryptoDetails from '../components/cryptoDetails';
import store from '../redux/store';

describe('Crypto Details', () => {
  test('render each crypto romponent', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <CryptoDetails />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
