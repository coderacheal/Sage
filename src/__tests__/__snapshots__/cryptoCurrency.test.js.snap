// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crypto Details render each crypto romponent 1`] = `
<div>
  <nav
    className="Nav"
  >
    <div>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        <i
          className="fa-solid fa-arrow-left"
        />
      </a>
    </div>
    <div>
      <h1
        className="brand"
      >
        SMART CRYPTO
      </h1>
    </div>
  </nav>
</div>
`;
